File: src/entity.cpp
#include "entity.hpp"

Entity::Entity(IPoint pos):
  position(pos)
{}

void Entity::setPosition(IPoint npos) { 
  position = npos; 
}

void Entity::move(IVector dir) {
  position += dir;
}

IPoint Entity::getPosition() {
  return position;
}

File: lib/entity.rb

require_relative 'geom'

class Entity
  attr_accessor :pos, :char, :color

  def initialize(pos, char, color)
    @pos = pos
    @char = char
    @color = color
  end

  def move(pos)
    @pos.x = pos.x
    @pos.y = pos.y
  end
end

